File: DevExpress-Demo/DevExpress.OutlookInspiredApp.Wpf/VB/DevExpress.OutlookInspiredApp.Wpf/Behaviors/SpellCheckerRichEditBehavior.vb
Imports System.Globalization
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.RichEdit
Imports DevExpress.Xpf.SpellChecker
Imports DevExpress.XtraRichEdit.SpellChecker
Imports DevExpress.XtraSpellChecker
Imports DevExpress.XtraSpellChecker.Native

Namespace DevExpress.DevAV

    Public Class SpellCheckerRichEditBehavior
        Inherits Behavior(Of RichEditControl)

        Private spellChecker As SpellChecker

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AssociatedObject.SpellChecker = spellChecker
        End Sub

        Public Sub New()
            spellChecker = CreateSpellChecker()
        End Sub

        Private Function CreateSpellChecker() As SpellChecker
            Dim spellChecker = New SpellChecker()
            spellChecker.Culture = New CultureInfo("en-US")
            spellChecker.SpellCheckMode = SpellCheckMode.AsYouType
            SpellCheckTextControllersManager.Default.RegisterClass(GetType(RichEditControl), GetType(RichEditSpellCheckController))
            Return spellChecker
        End Function
    End Class
End Namespace

File: DevExpress-Demo/DevExpress.OutlookInspiredApp.Wpf/CS/DevExpress.OutlookInspiredApp.Wpf/Behaviors/SpellCheckerRichEditBehavior.cs
using System.Globalization;
using System.IO;
using DevExpress.Internal;
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.RichEdit;
using DevExpress.Xpf.SpellChecker;
using DevExpress.XtraRichEdit.SpellChecker;
using DevExpress.XtraSpellChecker;
using DevExpress.XtraSpellChecker.Native;

namespace DevExpress.DevAV {
    public class SpellCheckerRichEditBehavior : Behavior<RichEditControl>{
        Xpf.SpellChecker.SpellChecker spellChecker;
        protected override void OnAttached() {
            base.OnAttached();
            AssociatedObject.SpellChecker = spellChecker;
        }
        public SpellCheckerRichEditBehavior() {
            spellChecker = CreateSpellChecker();
        }
        Xpf.SpellChecker.SpellChecker CreateSpellChecker() {
            var spellChecker = new Xpf.SpellChecker.SpellChecker();
            spellChecker.Culture = new CultureInfo("en-US");
            spellChecker.SpellCheckMode = XtraSpellChecker.SpellCheckMode.AsYouType;
            SpellCheckTextControllersManager.Default.RegisterClass(typeof(RichEditControl), typeof(RichEditSpellCheckController));
            return spellChecker;
        }
    }
}
